src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java

```diff
@@ -1,11 +1,834 @@
+        int splintNum = 1;
+        splitLabel.append(splintNum);
+        while (graph.containsVertexLabel(splitLabel.toString())) {
+            splintNum++;
+            splitLabel.replace(splitLabel.lastIndexOf("#") + 1, Integer.MAX_VALUE, String.valueOf(splintNum));
+        }
+        String vertexLabel = splitLabel.toString();
+        if (createSemiPermanentEdges) {
+            v = new SemiPermanentSplitterVertex(graph, vertexLabel, splitPoint.x, splitPoint.y, edge);
+        } else if (temporarySplit) {
+            v = new TemporarySplitterVertex(vertexLabel, splitPoint.x, splitPoint.y, edge, endVertex);
+        } else {
+            v = new SplitterVertex(graph, vertexLabel, splitPoint.x, splitPoint.y, edge);
+        }
+
+        // Split the 'edge' at 'v' in 2 new edges and connect these 2 edges to the
+        // existing vertices
+        P2<StreetEdge> edges = edge.split(v, destructiveSplitting, createSemiPermanentEdges);
+
+        if (destructiveSplitting || createSemiPermanentEdges) {
+            // update indices of new edges
+            synchronized (this) {
+                // Note: Write operations are not synchronized in HashGridSpatialIndex, hence the lock.
+                idx.insert(edges.first.getGeometry(), edges.first);
+                idx.insert(edges.second.getGeometry(), edges.second);
+
+                // if destructively splitting, remove the edge from the index
+                if (destructiveSplitting) {
+                    idx.remove(edge.getGeometry().getEnvelopeInternal(), edge);
+                }
+            }
+        }
+
+        if (destructiveSplitting) {
+            // remove original edge from the graph if split was destructive
+            edge.getToVertex().removeIncoming(edge);
+            edge.getFromVertex().removeOutgoing(edge);
+        }
+
+        return v;
+    }
+
+    /** Make the appropriate type of link edges from a vertex */
+    private void makeLinkEdges(Vertex from, StreetVertex to, final boolean destructiveSplitting) {
+        if (from instanceof TemporaryStreetLocation) {
+            makeTemporaryEdges((TemporaryStreetLocation) from, to, destructiveSplitting);
+        } else if (from instanceof TransitStop) {
+            makeTransitLinkEdges((TransitStop) from, to, destructiveSplitting);
+        } else if (from instanceof BikeRentalStationVertex) {
+            makeBikeRentalLinkEdges((BikeRentalStationVertex) from, to);
+        } else if (from instanceof BikeParkVertex) {
+            makeBikeParkEdges((BikeParkVertex) from, to, destructiveSplitting);
+        } else if (from instanceof CarRentalStationVertex) {
+            makeCarRentalLinkEdges((CarRentalStationVertex) from, to);
+        } else if (from instanceof VehicleRentalStationVertex) {
+            makeVehicleRentalLinkEdges((VehicleRentalStationVertex) from, to);
+        }
+    }
+
+    /** Make temporary edges to origin/destination vertex in origin/destination search **/
```

Comment:
Extra asterisk at end of java doc comment

Reply:
Will reformat.